lib/content-services/src/lib/category/services/category.service.spec.ts

```diff
@@ -71,4 +81,66 @@ describe('CategoryService', () => {
             expect(deleteSpy).toHaveBeenCalledOnceWith(fakeParentCategoryId);
         });
     }));
+
+    describe('searchCategories', () => {
+        const defaultMaxItems = 25;
+
+        let result: ResultSetPaging;
+
+        beforeEach(() => {
+            spyOnProperty(userPreferencesService, 'paginationSize').and.returnValue(defaultMaxItems);
+            result = new ResultSetPaging();
+            result.list = new ResultSetPagingList();
+            const category = new ResultSetRowEntry();
+            category.entry = new ResultNode();
+            category.entry.name = 'some name';
+            category.entry.path = new PathInfo();
+            category.entry.path.name = '/categories/General';
+            result.list.entries = [category];
+            spyOn(categoryService.searchApi, 'search').and.returnValue(Promise.resolve(result));
+        });
+
+        it('should call search on searchApi with correct parameters when specified all parameters', () => {
+            const name = 'name';
+            const skipCount = 10;
+            const maxItems = 100;
+
+            categoryService.searchCategories(name, skipCount, maxItems);
+            expect(categoryService.searchApi.search).toHaveBeenCalledWith({
+                query: {
+                    language: RequestQuery.LanguageEnum.Afts,
+                    query: `cm:name:"*${name}*" AND TYPE:'cm:category' AND PATH:"/cm:categoryRoot/cm:generalclassifiable//*"`
+                },
+                paging: {
+                    skipCount,
+                    maxItems
+                },
+                include: ['path']
+            });
+        });
+
+        it('should call search on searchApi with default parameters when skipped optional parameters', () => {
+            const name = 'name';
+
+            categoryService.searchCategories(name);
+            expect(categoryService.searchApi.search).toHaveBeenCalledWith({
+                query: {
+                    language: RequestQuery.LanguageEnum.Afts,
+                    query: `cm:name:"*${name}*" AND TYPE:'cm:category' AND PATH:"/cm:categoryRoot/cm:generalclassifiable//*"`
+                },
+                paging: {
+                    skipCount: 0,
+                    maxItems: defaultMaxItems
+                },
+                include: ['path']
+            });
+        });
+
+        it('should return observable which emits paging object for tags', (done) => {
```

Comment:
I think it should be for `categories` instead of `tags` :)

Reply:
done